### PR TITLE
Remove Journalbeat

### PR DESCRIPTION
**Pull request by kvch**
## What does this PR do?

This PR removes Journalbeat from Beats. The Beat has been experimental for years. Now it is released as a Filebeat input, so we no longer need the Beat.

## Why is it important?

Cleanup.

## Checklist

- [x] My code follows the style guidelines of this project
~~- [ ] I have commented my code, particularly in hard-to-understand areas~~
~~- [ ] I have made corresponding changes to the documentation~~
~~- [ ] I have made corresponding change to the default configuration files~~
~~- [ ] I have added tests that prove my fix is effective or that my feature works~~
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

### CODE REVIEW

File: x-pack/elastic-agent/pkg/agent/program/testdata/journal_config-journalbeat.yml [hunk 1]
```diff
@@ -1,15 +0,0 @@
-journalbeat:
```
Comment (ruflin) [inline x-pack/elastic-agent/pkg/agent/program/testdata/journal_config-journalbeat.yml:1]: Argh, we supported it even in Elastic Agent? @nimarezainia We likely need to mention this somewhere also in Elastic Agent 8.0 breaking changes that `log/journal` was removed.

**Review by ruflin — COMMENTED** [review]
Overall change LGTM. Lets get it into 8.0 and as a follow up lets see if we still have time to make it into 7.16.

CI is not happy including docs. I suspect there are a few dependencies. I would also not be suprised if the release manager also has some references to it and needs updating.

**Comment by cachedout** [comment]
>  I would also not be suprised if the release manager also has some references to it and needs updating.

There's a [sprinkling of references](https://github.com/elastic/infra/search?p=2&q=journalbeat) across the infra repo. Many of these look related to previous releases and shouldn't need to be touched, but a few many not be. I think that @elastic/infra-release probably need to be looped in here to take a look.

**Comment by kvch** [comment]
> > I would also not be suprised if the release manager also has some references to it and needs updating.
> 
> There's a [sprinkling of references](https://github.com/elastic/infra/search?p=2&q=journalbeat) across the infra repo. Many of these look related to previous releases and shouldn't need to be touched, but a few many not be. I think that @elastic/infra-release probably need to be looped in here to take a look.

I have already checked it. I mentioned in the original issue that the infra team has already removed Journalbeat from the release process, from download pages, etc. There is nothing else we have to do.

File: x-pack/elastic-agent/pkg/agent/program/testdata/journal_config-journalbeat.yml (hunk 1 above)
Comment (kvch) [reply x-pack/elastic-agent/pkg/agent/program/testdata/journal_config-journalbeat.yml:1]: I thought `journald` input was exposed under `log/journal`. Is that not correct?

Comment (ruflin) [reply x-pack/elastic-agent/pkg/agent/program/testdata/journal_config-journalbeat.yml:1]: The problem is that it is correct. Under what path do we expose the filebeat journal input?

**Comment by kvch** [comment]
jenkins run tests

**Comment by kvch** [comment]
jenkins run tests

**Comment by dedemorton** [comment]
@elasticmachine run elasticsearch-ci/docs

**Comment by kvch** [comment]
I have accidentally merged this PR. If you have any comments, I will address them in a follow-up PR.

**Pull request merged**